build-system/eslint-rules/resolve-inside-promise-resolver.js

```diff
@@ -86,30 +86,39 @@ module.exports = function(context) {
       }
 
       const {name} = resolve;
+      let assigned = false;
 
       if (resolver.type === 'ArrowFunctionExpression' &&
           resolver.expression === true) {
-        if (isResolveCall(resolver.body, name)) {
-          return;
-        }
+        const {body} = resolver;
+        assigned = isAssignment(body, name);
       } else {
         const {body} = resolver.body;
 
         for (let i = 0; i < body.length; i++) {
           const node = body[i];
-          if (node.type === 'ExpressionStatement' &&
-              isResolveCall(node.expression, name)) {
-            return;
+          if (node.type !== 'ExpressionStatement') {
+            continue;
+          }
+
+          const {expression} = node;
+          assigned = isAssignment(expression, name);
+          if (assigned) {
+            break;
           }
         }
       }
 
+      if (!assigned) {
+        return;
+      }
+
       const message = [
-        'Must call the resolve param.',
         'If you are creating a pending promise to extract a resolve function',
         'please use Deferred in the src/utils/promise.js module instead.',
```

Comment:
Can you suggest something based on your understanding? I'll definitely add a comment to the promise module, though.

Reply:
This message as written was hard for me to understand. However, it became clear after I read the comment you just added. How about something like this?
```
Instead of creating a pending Promise, use the Deferred class defined in src/utils/promise.js@61.
```

Reply:
Another source of confusion was the name of the rule: `resolve-inside-promise-resolver`. How about a simpler name that reflects what's going on here?  `no-promise-resolver`? `unnecessary-promise-resolver`?